content/articles/diogogpinto-filament-themes-full-guide.md

```diff
@@ -0,0 +1,333 @@
+
+```html
+<nav class="fi-breadcrumbs mb-2 hidden sm:block">
+```
+
+The `.fi-breadcrumbs` class is inserted to allow developers to further optimize the breadcrumbs' look and feel. By default, the class has no CSS properties defined in itself, it's only there for the developer to apply custom styles.
+
+#### Customizing the CSS
+
+Picking up in the above example of `.fi-breadcrumbs`, we can now apply custom CSS classes on our `theme.css`. You can either use Tailwind utility classes or just write vanilla CSS. Let's say you want to override the default margin `mb-2`. Inputting the below CSS will yield the exact same results:
+
+```css
+.fi-breadcrumbs {
+    @apply mb-4;
+}
+
+.fi-breadcrumbs {
+    margin-bottom: 1rem;
+}
+```
+
+As the `.fi-breadcrumbs` class is inserted before the default Tailwind utility classes, applying custom CSS will always override the default style.
+
+#### Diving deeper
+
+If we dive deeper into the breadcrumbs, we will find all the following classes, in this hierarchy:
+
+```
+- .fi-breadcrumbs
+-- .fi-breadcrumbs-list
+--- .fi-breadcrumbs-item
+---- .fi-breadcrumbs-item-separator
+---- .fi-breadcrumbs-item-label
+```
+
+Let's say we want to really change the breadcrumbs appearance. We can style each of this classes. Check the following example:
+
+```css
+.fi-breadcrumbs-list {
+    @apply gap-x-0;
+}
+
+.fi-breadcrumbs-item {
+    @apply gap-x-0;
+}
+
+.fi-breadcrumbs-item-label {
+    @apply font-thin dark:text-primary-500 text-primary-500;
+}
+
+span.fi-breadcrumbs-item-label {
+    @apply border-b-2 border-b-primary-500 font-semibold -mb-[2px];
+}
+
+.fi-breadcrumbs-item-separator {
+    @apply fill-primary-500 h-4;
+}
+```
+
+![Breadcrumbs Final Example](/images/content/articles/diogogpinto-filament-themes-full-guide/breadcrumbs-final.webp)
+
+There are a lot of classes around every Filament component. You can go through each one and apply your custom CSS, so you can create a unique design for your Panel.
+
+> [!TIP]
+> There's a useful Filament Plugin ([Theme Inspector](https://filamentphp.com/plugins/codewithdennis-theme-inspector)) that lets you inspect all the classes in your panel in your dev environment. You can even copy to clipboard! It really smoothes the development process.
+
+### Going behind CSS
+
+Sometimes there are some things your theme requires that go behind CSS. You may want to add a custom Livewire component, Blade View or JS script. As these modifications are outside the scope of this guide, I would advise you to check the following resources:
+
+- [Render Hooks on the Official Docs](https://filamentphp.com/docs/3.x/support/render-hooks) - check how you can render custom views almost anywhere in your panels
+- [Registering Assets](https://filamentphp.com/docs/3.x/support/assets#registering-javascript-files) - check how you can register custom JS/CSS into your Filament panels
+
+### The ~wrong~ unrecommended way to customize your theme
```

Comment:
We really should not have this section in my opinion as this approach should basically never be used.

Reply:
I'm on the fence about this one. On the one hand, I completely agree that we almost never want this approach to be used. However, on the other hand, by having this here, we're specifically calling out the fact that this is the _wrong_ approach, dissuading people from using it.

I guess it comes down to whether or not this is going to prompt more people to remember that this is an option and use it or if it's going to prevent people from using it because it's explicitly stated not to.

I think I come down more on the side of leave it in to ensure that people know it's the wrong way to do things. My hunch is that the people reading this article are (in general) going to be deep enough in Filament to know that overriding views is an option in Laravel 🤷🏻

Reply:
I can see both points!

I inserted because sometimes you may want to change specific functionality around whole components that may not necessarily break functionality on updates. My case was the topbar, but later I disabled it and used render hooks. But that approach gave me some headaches around mobile and had to create a whole new mobile trigger for the sidebar - if I had to do it all over again, I would publish the file and monitor each release for breaking changes.